src/types.ts

```diff
@@ -9,7 +9,7 @@ export interface Project {
   license: string;
   tags: string[];
   codeUrl: string;
-  demoUrl?: string | undefined | null;
+  demoUrl: string | null;
```

Comment:
I'm puzzled. Isn't `null` used in cases when sth is deliberately of error/wrong value and is replaced with `null` instead? And `undefined` is simply where the value is optional. So in case of licenses and demo URLs, those fields are not mandatory, so they should be left out from the projects.json (and undefined when reading them in the code).

Reply:
There are some conventions that `undefined` means "Not Applicable", and `null` if it approximates Maybe/Option type. But I don't think those matter

I chose `null` here because I would rather have all optional fields visible, rather than removed. Some usecases:
- `{ "name": "BlackSea", "slug": "blacksea", "demoUrl": null }` reminds you that you might have missed it
- you can grep for `"demoUrl": null` to count projects with a missing field
- when we add a new field to yaml, and only a few projects have it: implementing displaying it on oasis.net/ecosystem you don't have to read whole projects.json or recent PRs to find its name